libraries/helpers.rb

```diff
@@ -23,24 +23,29 @@ module Helpers
 
     def psql_command_string(new_resource, query, grep_for: nil, value_only: false)
       cmd = "/usr/bin/psql -c \"#{query}\""
-      cmd << " -d #{new_resource.database}" if new_resource.database
-      cmd << " -U #{new_resource.user}"     if new_resource.user
+      cmd << " -d #{new_resource.database}" if new_resource.respond_to?(:database) && new_resource.database
+      cmd << " -U #{new_resource.user}"
       cmd << " --host #{new_resource.host}" if new_resource.host
-      cmd << " --port #{new_resource.port}" if new_resource.port
+      cmd << " --port #{new_resource.port}"
       cmd << ' --tuples-only'               if value_only
       cmd << " | grep #{grep_for}"          if grep_for
       cmd
     end
 
     def execute_sql(new_resource, query)
-      # If we don't pass in a user to the resource
-      # default to the postgres user
-      user = new_resource.user || 'postgres'
+      # If we don't pass in a user to the resource, set user to postgres
+      # If we are connecting to a remote server then we don't need to execute the command as a particular user
```

Comment:
As chef run as root `psql` will run as root?

Reply:
Yes it will. Is there any other choice ? The `postgres` user may not be present.

Reply:
The cookbook user can set `nobody`?

Reply:
Yes it could but that becomes tricky to implement : 
- use new_resource.user preferably
- else use `postgres`
- but if `postgres` doesn't exist then use `nobody`

I'm not sure it's worth the effort.

Reply:
Ok keep it simple stupid, plus on some distro `nobody` may not be available